src/main/java/edu/arizona/kra/protocol/onlinereview/dao/ProtocolOnlineReviewDaoOjb.java

```diff
@@ -0,0 +1,152 @@
+        List<ProtocolOnlineReviewBase> searchResults = new ArrayList();
+        searchResults = (List<ProtocolOnlineReviewBase>) getPersistenceBrokerTemplate().getCollectionByQuery(query);
+        return searchResults;
+    }
+
+
+    public List<ProtocolOnlineReviewBase> getProtocolOnlineReviewsByQuery(QueryByCriteria query) {
+        List<ProtocolOnlineReviewBase> retVal = null;
+        logQuery(query);
+        retVal = (List<ProtocolOnlineReviewBase>) getPersistenceBrokerTemplate().getCollectionByQuery(query);
+
+        return retVal;
+    }
+
+
+    public List<ProtocolOnlineReviewBase> getProtocolOnlineReviewsByReportQuery(ReportQueryByCriteria query) {
+        List<ProtocolOnlineReviewBase> retVal = null;
+
+        logQuery(query);
+        retVal = (List<ProtocolOnlineReviewBase>) getPersistenceBrokerTemplate().getCollectionByQuery(query);
+
+        return retVal;
+    }
+
+
+    private static void logQuery(Query q) {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug(q.toString());
+        }
+    }
+
+
+    public List<ProtocolOnlineReviewBase> getCustomSearchResults() {
+        return getCustomSearchResults();
+    }
+
+
+    @Override
+    public List<ProtocolOnlineReviewBase> getCustomSearchResults(Map<String, String> fieldValues) {
+
+        List<ProtocolOnlineReviewBase> searchResults = new ArrayList();
+
+        Long matchingResultsCount = null;
+
+        ProtocolOnlineReview protocolOnlineReview = new ProtocolOnlineReview();
+
+        Criteria criteria1 = this.getCollectionCriteriaFromMap( protocolOnlineReview, fieldValues);
+
+        if (!((String)fieldValues.get("dateDue")).isEmpty() &&
+                ((String)fieldValues.get("dateDue")).startsWith("<=")) {
+            String dateDueString = (fieldValues.get("dateDue")).substring(2);
+            criteria1.addOrderByDescending("dateDue");
+        }
+
+        if (!((String)fieldValues.get("dateRequested")).isEmpty() &&
+                ((String)fieldValues.get("dateRequested")).startsWith("<=")) {
+            String dateRequestedString = (fieldValues.get("dateRequested")).substring(2);
+            criteria1.addOrderByDescending("dateRequested");
+        }
+
+        Integer searchResultsLimit = LookupUtils.getSearchResultsLimit(ProtocolOnlineReview.class);
+
+        if (searchResultsLimit != null && searchResultsLimit >= 0) {
+            matchingResultsCount = new Long((long)this.getPersistenceBrokerTemplate().getCount(QueryFactory.newReportQuery(ProtocolOnlineReview.class, criteria1)));
+            org.kuali.rice.krad.lookup.LookupUtils.applySearchResultsLimit(ProtocolOnlineReview.class, criteria1, this.getDbPlatform(), searchResultsLimit);
+        }
+
+        if (matchingResultsCount == null || matchingResultsCount.intValue() <= searchResultsLimit) {
+            matchingResultsCount = new Long(0L);
+        }
+
+        // Below Code Adds Logic performed by filterresults()
```

Comment:
Create a helper method and call it for the two queries. Explain why we need two.